### ISSUE TITLE
Investigate Date Formatting Issues

### ISSUE THREAD

**Issue by thomersch** [thomersch/openstreetmap-calendar#42]
The whole date range formatting is kind of a faff: The templates are difficult to read and the i10n situation is a little bit complex. But besides the conventions, there are some proper bugs:

* For fr-(FR|BE) it does not capitalize the months: "12 octobre"
* For de-AT it does say "Januar" instead of "Jänner"

At least those two need to be investigated and fixed. I assume the Django API is being misused in OSMCAL code or Django itself has some issues.

**Comment by thomersch** [comment]
Apparently, the french capitalization issue is fixed by using "E" instead of "F" in the date template (fixed in https://github.com/thomersch/openstreetmap-calendar/commit/e9359f92635f43116ef003bbcc47c460e9bfa6f5).

**Comment by thomersch** [comment]
We should consider using [Babel](http://babel.pocoo.org/en/latest/) for formatting dates, since it seems to have the functionality to format date/time ranges.

**Comment by assanges** [comment]
> We should consider using [Babel](http://babel.pocoo.org/en/latest/) for formatting dates, since it seems to have the functionality to format date/time ranges.

Agreed. YMD format is being used instead in some Languages (e.g. dz, en-CA/fr-CA, hu, lt, ja, ko, mn, zh, etc), so they are incompatible with current syntax.

**Comment by thomersch** [comment]
I have done a little bit in that direction (e.g. "2021年4月22日 12:00 德国（柏林）时间"), but Babel still needs some glue code if we want to use it. A particular problem are multi-day events with times, e.g. "20 Aug 2029, 17:00:00 – 22 Aug 2029, 19:00:00 UTC", where neither month formatting nor time formatting are well-controllable. For detail pages I would want to show full-width months (e.g. "August"), but skip the seconds altogether. This will require some more hacking around, but I haven't given up yet!
